presto-main/src/main/java/com/facebook/presto/operator/GroupIdOperator.java

```diff
@@ -44,22 +46,24 @@
         private final List<Type> inputTypes;
         private final List<Type> outputTypes;
         private final List<List<Integer>> groupingSetChannels;
+        private final Map<Integer, Integer> passthroughChannelMapping;
 
         private boolean closed;
 
         public GroupIdOperatorFactory(
                 int operatorId,
                 PlanNodeId planNodeId,
                 List<? extends Type> inputTypes,
-                List<List<Integer>> groupingSetChannels)
+                List<? extends Type> outputTypes,
+                List<List<Integer>> groupingSetChannels,
+                Map<Integer, Integer> passthroughChannelMapping)
```

Comment:
Right, the only one that needs to know is the LocalExecutionPlanner. It could be part of the contract of this operator that channels are laid out in this order: [groupId, grouping columns, everything else]